src/simplay-web/src/event/InteractionLine.ts

```diff
@@ -0,0 +1,98 @@
+import { DisplayEntity } from '../Entity';
+import * as PIXI from 'pixi.js';
+import { SimplayContext } from '../SimplayContext';
+
+export class InteractionLine {
+  public readonly graphic: PIXI.Graphics;
+  private context: SimplayContext;
+  private offsetX = 0;
+  private offsetY = 0;
+  private switch = false;
+  private ticker: PIXI.Ticker;
+  constructor(
+    private sourceEntity: DisplayEntity,
+    private targetEntity: DisplayEntity,
+    context: SimplayContext
+  ) {
+    this.graphic = new PIXI.Graphics();
+    this.graphic.name = `${sourceEntity.container.name} -> ${targetEntity.container.name}`;
+    this.context = context;
+    this.redraw();
+    this.ticker = this.context.app.ticker.add(() => {
+      this.redraw();
+    });
+    this.context.interactionContainer.addChild(this.graphic);
+  }
+
+  public destroy() {
+    this.context.interactionContainer.removeChild(this.graphic);
+    this.ticker.destroy();
+    this.graphic.destroy();
+  }
+
+  redraw() {
+    this.graphic.clear();
+    this.createGradientLine(
+      this.sourceEntity.container.x + this.sourceEntity.container.width / 2,
+      this.sourceEntity.container.y + this.sourceEntity.container.height / 2,
+      this.targetEntity.container.x + this.targetEntity.container.width / 2,
+      this.targetEntity.container.y + this.targetEntity.container.height / 2,
+      21,
+      0x000000,
+      0xffffff
+    );
+  }
+
+  createGradientLine(
```

Comment:
I had hard times to understand this method, could we split it in multiple functions, to make it better understandable? just an idea not sure if possible